app/routers/reset_password.py

```diff
@@ -0,0 +1,135 @@
+from typing import Optional
+
+from fastapi import APIRouter, Depends, Request
+from pydantic import ValidationError
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+from starlette.status import HTTP_302_FOUND
+
+from app.dependencies import get_db, templates
+from app.internal.security.ouath2 import (
+    is_email_compatible_to_username,
+    get_jwt_token,
+    create_jwt_token,
+    update_password,
+)
+from app.internal.email import BackgroundTasks, send_reset_password_mail
+from app.internal.security.schema import ForgotPassword, ResetPassword
+from app.routers.login import router as login_router
+
+
+router = APIRouter(
+    prefix="",
+    tags=["/reset_password"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/forgot-password")
+async def forgot_password_form(request: Request) -> templates:
+    """rendering forgot password form get method"""
+    return templates.TemplateResponse(
+        "forgot_password.html",
+        {
+            "request": request,
+        },
+    )
+
+
+@router.post("/forgot-password")
+async def forgot_password(
+    request: Request,
+    background_tasks: BackgroundTasks,
+    db: Session = Depends(get_db),
+) -> templates:
+    """
+    Validaiting form data fields.
+    Validaiting form data against database records.
+    If all validations succeed, creating jwt token,
+    then sending email to the user with a reset password route link.
+    The contains the verafiction jwt token.
+    """
+    form = await request.form()
+    form_dict = dict(form)
+    try:
+        # validating form data by creating pydantic schema object
+        user = ForgotPassword(**form_dict)
+    except ValidationError:
+        user = None
+    if user:
+        user = await is_email_compatible_to_username(db, user)
+    if not user:
+        return templates.TemplateResponse(
+            "forgot_password.html",
+            {"request": request, "message": "Please check your credentials"},
+        )
+    user.email_verification_token = create_jwt_token(user, jwt_min_exp=15)
+    await send_reset_password_mail(user, background_tasks)
+    return templates.TemplateResponse(
+        "forgot_password.html",
+        {
+            "request": request,
+            "message": "Email for reseting password was sent",
+        },
+    )
+
+
+@router.get("/reset-password")
+async def reset_password_form(
+    request: Request,
+    email_verification_token: Optional[str] = "",
```

Comment:
That's not a good idea for many reasons.

1. GET requests saved in the history, in the server's logs and in the browser. Sending the JWT using get could help 3rd party malicious player to get the token and impersonate someone without him/her knowing it.
2. The same goes for sending the JWT via unencrypted mail.

Reply:
The token expiration time is 15 minutes. What would you suggest me to do then? that the main logic of this feature